lib/eblox/data/taxonomy.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Eblox.Data.Taxonomy do
+  @moduledoc false
+
+  use Supervisor
+
+  @type post_id() :: term()
+  @type registry_option() :: Registry.start_option()
+
+  @callback registry_options([registry_option()]) :: [registry_option()]
+  @callback on_add(module(), post_id(), term()) :: :ok | :error
+  @callback on_remove(module(), post_id()) :: :ok
+
+  defmodule Meta do
+    @moduledoc false
+
+    use Agent
+
+    alias Eblox.Data.Taxonomy.Meta
+
+    @type t :: %{
+      __struct__: Meta,
+      impl: module(),
+    }
+    @enforce_keys ~w|impl|a
+    defstruct ~w|impl|a
+
+    def start_link([{:impl, impl} | opts]) do
+      Agent.start_link(fn -> %Meta{impl: impl} end, opts)
+    end
+
+    def impl(name) do
+      Agent.get(name, &(&1 |> Map.get(:impl)))
+    end
+  end
+
+  def start_link(opts), do:
+    Supervisor.start_link(__MODULE__, opts)
+
+  def init(opts \\ []) do
```

Comment:
In what scenario you’d expect this to be called without arguments?

```suggestion
  def init(opts) do
```